packages/context/src/context.ts

```diff
@@ -797,21 +799,27 @@ export class Context extends EventEmitter {
   ): ValueOrPromise<ValueType | undefined> {
     const {key, propertyPath} = BindingKey.parseKeyWithPath(keyWithPath);
 
-    optionsOrSession = asResolutionOptions(optionsOrSession);
-
-    const binding = this.getBinding<ValueType>(key, optionsOrSession);
-    if (binding == null) return undefined;
+    const options = asResolutionOptions(optionsOrSession);
+
+    let binding: Binding<ValueType> | undefined;
+    try {
+      binding = this.getBinding<ValueType>(key, options);
+      if (binding == null) return undefined;
+    } catch (err) {
+      throw new ResolutionError(err, {
```

Comment:
I find this approach problematic because it hides the part of the original stack trace that's pointing into `getBinding` internals and possibly into user-provided resolver.

Consider the following code:

```js
app.bind('foo').toDynamicValue(function getFoo() {
  throw new Error('something happened');
});
```

When trying to resolve `foo`, we should preserve the stack frames pointing to `getFoo`.

Reply:
Please note that I intentionally only catch the error thrown from `getBinding()` (not including the error from `getValue()`.